### ISSUE TITLE
Subclasses of product

### ISSUE THREAD

**Issue by kristiandaugaard** [silvershop/silvershop-stock#9]
If you subclass Product, `hasAvailableStock` will continue to return false, because `ProductStockExtension::getStockBaseIdentifier()` only assumes 'Product' or 'ProductVariation.

This seems to solve it for me:
```
public function getStockBaseIdentifier()
{
    return ($this->owner instanceof SiteTree) ? $this->owner->ClassName : "ProductVariation";
}
```

PR here #10

**Comment by wilr** [comment]
Sounds like you're not saving the product relation under `ProductID` even if you subclass Product to say `MyAppProduct` the $orderItem->ProductID should point to `MyAppProduct`. That's the purpose of that method. Saving it to anything else is going to cause all sorts of other issues with silvershop since `->product()` is used throughout.

**Comment by kristiandaugaard** [comment]
What you say makes sense. I dont know why mine is acting up.
My subclass of product doesn't really change much, just adds some fields to getCMSFields and the db array. the `hasAvailableStock` method will return false even if stock is added. 

Might do other tests in the future, but due to deadlines i may just use my fork for now.

**Comment by wilr** [comment]
Fixed with #14
